app/components/gutenberg/Block/BlockStyle.tsx

```diff
@@ -63,24 +63,15 @@ export const BlockStyle = ({
       <style jsx>{`
         .${className} {
           ${generateStyles(block, "xs")}
-        }
-
-        ${
-          /* @TODO: Find a way to make the class name scoped, see #11 */
-          block?.attributes?.styles?.xs?.custom &&
-          block?.attributes?.styles?.xs?.custom.length
-            ? block.attributes.styles.xs.custom.replace(
-                "selector",
-                `.${className}`
-              )
-            : ``
+          ${extractStyles(block?.attributes?.styles?.xs?.custom)}
         }
 
         @media (min-width: ${token[
           screens.sm.antdToken as keyof typeof token
         ]}px) {
           .${className} {
             ${generateStyles(block, "sm")}
+            ${extractStyles(block?.attributes?.styles?.sm?.custom)}
```

Comment:
@Danishkhurshid This isn't valid CSS, please test first. Shouldn't have a selector manually defined.